frontend/packages/console-plugin-sdk/src/codegen/active-plugins.ts

```diff
@@ -13,8 +15,56 @@ import { Extension, ActivePlugin } from '../typings';
 import { trimStartMultiLine } from '../utils/string';
 import { consolePkgScope, PluginPackage } from './plugin-resolver';
 
+const getExtensionsFilePath = (pkg: PluginPackage) => path.resolve(pkg._path, extensionsFile);
+
+/**
+ * Guess the file path of the module (e.g., the extension,
+ * any barrel/index file) based on the given base path.
+ *
+ * Returns the base path if no file is found.
+ */
+const guessModuleFilePath = (basePath: string) => {
+  const extensions = ['.tsx', '.ts', '.jsx', '.js'];
+
+  // Sometimes the module is an index file, but the exposed module path only specifies the directory.
+  // In that case, we need an explicit check for the index file.
+  const indexModulePaths = ['index.ts', 'index.js'].map((i) => path.resolve(basePath, i));
+
+  const pathsToCheck = [...indexModulePaths, ...extensions.map((ext) => `${basePath}${ext}`)];
+
+  for (const p of pathsToCheck) {
+    if (fs.existsSync(p)) {
+      return p;
+    }
+  }
+
+  // A file couldn't be found, return the original module path. A compilation warning
+  // may be pushed by `getActivePluginsModuleData`
+  return basePath;
+};
+
+const getExposedModuleFilePath = (pkg: PluginPackage, moduleName: string) => {
+  const modulePath = path.resolve(pkg._path, pkg.consolePlugin.exposedModules[moduleName]);
+
+  // Check if there is a file extension (no extra guessing needed)
+  if (!path.extname(modulePath)) {
```

Comment:
:+1: for skipping the guessing when there is an extension present.

I'd personally use a ternary operator here (very simple one):
```ts
return path.extname(modulePath)
  ? modulePath // Path already contains a file extension
  : guessModuleFilePath(modulePath);
```